### PR TITLE
Problem: (CRO-527) Punishment information is not reflected in staked state

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added punishment information in staked state

### CODE REVIEW

File: chain-abci/src/slashing.rs [hunk 1]
```diff
@@ -11,23 +12,31 @@ pub struct SlashingSchedule {
...
     }
 
     /// Updates slash ratio (only if proposed ratio is greater than current)
     #[inline]
-    pub fn update_slash_ratio(&mut self, new_ratio: SlashRatio) {
+    pub fn update_slash_ratio(&mut self, new_ratio: SlashRatio, punishment_kind: PunishmentKind) {
         if new_ratio > self.slash_ratio {
-            self.slash_ratio = new_ratio
+            self.slash_ratio = new_ratio;
+            self.punishment_kind = punishment_kind;
```
Comment (tomtau) [inline chain-abci/src/slashing.rs]: seems this isn't tested -- perhaps add a test where two infractions happen?

File: chain-core/src/init/config.rs [hunk 2]
```diff
@@ -54,6 +55,9 @@ impl fmt::Display for DistributionError {
             DistributionError::InvalidVotingPower => {
                 write!(f, "Invalid voting power")
             },
+            DistributionError::InvalidPunishmentConfiguration => {
+                write!(f, "Invalid punishment configuration (maybe slash_wait_period < jail_duration)")
```
Comment (calvinaco) [inline chain-core/src/init/config.rs]: This message seems to imply that the `slash_wait_period` should not be less than `jail_duration`; I think it should give some hinting words to describe you should make `slash_wait_period < jail_duration`

File: chain-abci/src/slashing.rs (hunk 1 above)
Comment (devashishdxt) [reply chain-abci/src/slashing.rs]: Added a test for checking multiple validator faults

File: chain-core/src/init/config.rs (hunk 2 above)
Comment (devashishdxt) [reply chain-core/src/init/config.rs]: Changed

**Pull request merged**